### PR TITLE
update build_taglib.bat for cmake use

### PR DESCRIPTION
**Pull request by daschuer**

### CODE REVIEW

File: build_taglib.bat [hunk 1]
```diff
@@ -32,50 +32,25 @@ mkdir %PLATFORM%
 cd %PLATFORM%
 REM NOTE(pegasus): generated solution with
```
Comment (rryan) [inline build_taglib.bat:34]: taglib doesn't rely on Qt so I don't think the `CMAKE_PREFIX_PATH` change is necessary?

Comment (daschuer) [reply build_taglib.bat:34]: No, we can remove this.

File: build_taglib.bat [hunk 2]
```diff
@@ -32,50 +32,25 @@ mkdir %PLATFORM%
...
 
-"%CMAKEDIR%\cmake" ../ -G %COMPILERPROJECT% -DBUILD_SHARED_LIBS=%DYNAMIC% -DCMAKE_SUPPRESS_REGENERATION=1 -DCMAKE_USE_RELATIVE_PATHS=1 -DZLIB_INCLUDE_DIR=%INCLUDE_DIR% -DZLIB_LIBRARY=%LIB_DIR%\%ZLIBNAME%
+"%CMAKEDIR%\cmake" .. -G %COMPILERPROJECT% -DBUILD_SHARED_LIBS=%DYNAMIC% -DCMAKE_SUPPRESS_REGENERATION=1 -DZLIB_INCLUDE_DIR=%INCLUDE_DIR% -DZLIB_LIBRARY=%LIB_DIR%\%ZLIBNAME% -DCMAKE_PREFIX_PATH=%ROOT_DIR%\Qt-%QT_VERSION%\lib\cmake -DCMAKE_INSTALL_PREFIX=%ROOT_DIR% -DCMAKE_INSTALL_INCLUDEDIR=%INCLUDE_DIR%
```
Comment (rryan) [inline build_taglib.bat]:
```
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_INSTALL_INCLUDEDIR
```

Looks like `CMAKE_INSTALL_INCLUDEDIR` is unused. Remove?

File: build_taglib.bat [hunk 3]
```diff
@@ -32,50 +32,25 @@ mkdir %PLATFORM%
...
-%XCOPY% taglib\toolkit\*.tcc %INCLUDE_DIR%\taglib
-%XCOPY% taglib\trueaudio\*.h %INCLUDE_DIR%\taglib
-%XCOPY% taglib\wavpack\*.h %INCLUDE_DIR%\taglib
-%XCOPY% taglib\xm\*.h %INCLUDE_DIR%\taglib
+"%CMAKEDIR%\cmake" --build . --target install 
+IF ERRORLEVEL 1 (
+    SET VALRETURN=1
+    goto END
+)
+
```
Comment (rryan) [inline build_taglib.bat:52]: please use %CONFIG%

File: build_taglib.bat [hunk 4]
```diff
@@ -32,50 +32,25 @@ mkdir %PLATFORM%
...
-%XCOPY% taglib\riff\*.h %INCLUDE_DIR%\taglib
-%XCOPY% taglib\riff\aiff\*.h %INCLUDE_DIR%\taglib
-%XCOPY% taglib\riff\wav\*.h %INCLUDE_DIR%\taglib
-%XCOPY% taglib\s3m\*.h %INCLUDE_DIR%\taglib
-%XCOPY% taglib\toolkit\*.h %INCLUDE_DIR%\taglib
-%XCOPY% taglib\toolkit\*.tcc %INCLUDE_DIR%\taglib
-%XCOPY% taglib\trueaudio\*.h %INCLUDE_DIR%\taglib
-%XCOPY% taglib\wavpack\*.h %INCLUDE_DIR%\taglib
-%XCOPY% taglib\xm\*.h %INCLUDE_DIR%\taglib
+"%CMAKEDIR%\cmake" --build . --target install
```
Comment (rryan) [inline build_taglib.bat]: Does this need a `--config %CONFIG%`? Looking at the build logs, this seems to trigger build in CONFIG=Debug mode as well.

Comment (rryan) [reply build_taglib.bat]: Looks like it's installing the Debug version:
```
  -- Install configuration: "Debug"
  -- Installing: C:/jenkins/bs-2.2-win-prs/amd64-ReleaseFastbuild/bin/taglib-config.cmd
  -- Installing: C:/jenkins/bs-2.2-win-prs/amd64-ReleaseFastbuild/lib/pkgconfig/taglib.pc
  -- Installing: C:/jenkins/bs-2.2-win-prs/amd64-ReleaseFastbuild/lib/tag.lib
  -- Installing: C:/jenkins/bs-2.2-win-prs/amd64-ReleaseFastbuild/include/taglib/tag.h
  -- Installing: C:/jenkins/bs-2.2-win-prs/amd64-ReleaseFastbuild/include/taglib/fileref.h
  -- Installing: C:/jenkins/bs-2.2-win-prs/amd64-ReleaseFastbuild/include/taglib/audioproperties.h
  -- Installing: C:/jenkins/bs-2.2-win-prs/amd64-ReleaseFastbuild/include/taglib/taglib_export.h
  -- Installing: C:/jenkins/bs-2.2-win-prs/amd64-ReleaseFastbuild/include/taglib/taglib_config.h
  -- Installing: C:/jenkins/bs-2.2-win-prs/amd64-ReleaseFastbuild/include/taglib/taglib.h
  -- Installing: C:/jenkins/bs-2.2-win-prs/amd64-ReleaseFastbuild/include/taglib/tstring.h
  -- Installing: C:/jenkins/bs-2.2-win-prs/amd64-ReleaseFastbuild/include/taglib/tlist.h
  -- Installing: C:/jenkins/bs-2.2-win-prs/amd64-ReleaseFastbuild/include/taglib/tlist.tcc
  -- Installing: C:/jenkins/bs-2.2-win-prs/amd64-ReleaseFastbuild/include/taglib/tstringlist.h
  -- Installing: C:/jenkins/bs-2.2-win-prs/amd64-ReleaseFastbuild/include/taglib/tbytevector.h
...
```

Comment (daschuer) [reply build_taglib.bat]: Ah, right. I will fix that.

File: build_taglib.bat (hunk 2 above)
Comment (rryan) [inline build_taglib.bat]: Could you edit `build_environment.bat` to set a `PLATFORM_TOOLSET` environment variable to `v%VCVERSION%_xp` and then set it via `-T %PLATFORM_TOOLSET%` here?

Comment (daschuer) [reply build_taglib.bat]: Ah yes, thats it.

**Comment by daschuer** [comment]
done.

**Comment by Be-ing** [comment]
The build succeeded and the environment has the previously missing taglib_config.h file now. Thank you!

File: build_taglib.bat (hunk 3 above)
Comment (rryan) [reply build_taglib.bat:52]: This didn't get done (so it wouldn't work for debug mode), and it looks like the PDB path wasn't quite right.

Fixed (hopefully) in https://github.com/mixxxdj/buildserver/commit/729dc209e64744246f7650cb7fccacc23f58d8d8

Comment (Be-ing) [reply build_taglib.bat:52]: Thanks, I aborted the old build. Now the build for 729dc20 is going.

**Pull request merged**